orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfiguration.java

```diff
@@ -0,0 +1,224 @@
+  "com.netflix.spinnaker.orca.kato.tasks"
+})
+@EnableConfigurationProperties({
+  CloudDriverConfigurationProperties.class,
+  PollerConfigurationProperties.class
+})
+@Slf4j
+class CloudDriverConfiguration {
+
+  @ConditionalOnMissingBean(ObjectMapper.class)
+  @Bean
+  ObjectMapper mapper() {
+    return OrcaObjectMapper.newInstance();
+  }
+
+  @Bean
+  ClouddriverRetrofitBuilder clouddriverRetrofitBuilder(
+      ObjectMapper objectMapper,
+      OkHttpClientProvider clientProvider,
+      RestAdapter.LogLevel retrofitLogLevel,
+      RequestInterceptor spinnakerRequestInterceptor,
+      CloudDriverConfigurationProperties cloudDriverConfigurationProperties) {
+    return new ClouddriverRetrofitBuilder(
+        objectMapper,
+        clientProvider,
+        retrofitLogLevel,
+        spinnakerRequestInterceptor,
+        cloudDriverConfigurationProperties);
+  }
+
+  static class ClouddriverRetrofitBuilder {
+    ObjectMapper objectMapper;
+    OkHttpClientProvider clientProvider;
+    RestAdapter.LogLevel retrofitLogLevel;
+    RequestInterceptor spinnakerRequestInterceptor;
+    CloudDriverConfigurationProperties cloudDriverConfigurationProperties;
+
+    ClouddriverRetrofitBuilder(
+        ObjectMapper objectMapper,
+        OkHttpClientProvider clientProvider,
+        RestAdapter.LogLevel retrofitLogLevel,
+        RequestInterceptor spinnakerRequestInterceptor,
+        CloudDriverConfigurationProperties cloudDriverConfigurationProperties) {
+      this.objectMapper = objectMapper;
+      this.clientProvider = clientProvider;
+      this.retrofitLogLevel = retrofitLogLevel;
+      this.spinnakerRequestInterceptor = spinnakerRequestInterceptor;
+      this.cloudDriverConfigurationProperties = cloudDriverConfigurationProperties;
+    }
+
+    <T> T buildWriteableService(Class<T> type) {
+      return buildService(type, cloudDriverConfigurationProperties.getCloudDriverBaseUrl());
+    }
+
+    private <T> T buildService(Class<T> type, String url) {
+      return new RestAdapter.Builder()
+          .setRequestInterceptor(spinnakerRequestInterceptor)
+          .setEndpoint(newFixedEndpoint(url))
+          .setClient(
+              new Ok3Client(
+                  clientProvider.getClient(new DefaultServiceEndpoint("clouddriver", url))))
+          .setLogLevel(retrofitLogLevel)
+          .setLog(new RetrofitSlf4jLog(type))
+          .setConverter(new JacksonConverter(objectMapper))
+          .build()
+          .create(type);
+    }
+
+    private <T> SelectableService buildReadOnlyService(Class<T> type) {
+      List<String> urls =
+          cloudDriverConfigurationProperties.getCloudDriverReadOnlyBaseUrls().stream()
+              .map(CloudDriverConfigurationProperties.BaseUrl::getBaseUrl)
+              .collect(Collectors.toList());
+
+      if (urls.isEmpty()
+          || urls.stream()
+              .allMatch(
+                  url ->
+                      url.equalsIgnoreCase(
```

Comment:
The groovy code is:
```
      if (cloudDriverConfigurationProperties.cloudDriverReadOnlyBaseUrls*.baseUrl == [cloudDriverConfigurationProperties.cloudDriverBaseUrl]) {
```
and `==` is a case-sensitive comparison, so:
```suggestion
                      url.equals(
```
?